auslib/web/admin/views/mapper.py

```diff
@@ -0,0 +1,76 @@
+from auslib.web.admin.views.csrf import CSRFView
+from auslib.web.admin.views.rules import RulesAPIView
+from auslib.web.admin.views.permissions import UsersView
+from jsonschema import draft4_format_checker
+from jsonschema.compat import str_types
+from wtforms.validators import ValidationError
+from auslib.util.comparison import get_op
+from auslib.util.versions import MozillaVersion
+import operator
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def csrf_get():
+    return CSRFView().get()
+
+
+def rules_get():
+    return RulesAPIView().get()
+
+
+def rules_post():
+    return RulesAPIView().post()
+
+
+def users_get():
+    return UsersView().get()
+
+
+@draft4_format_checker.checks(format="buildID", raises=ValidationError)
+def operator_validator(field_value):
+    log.debug('starting in operator_validator: buildID is %s' % field_value)
+    if not isinstance(field_value, str_types):
+        return True
+    # empty input is fine
+    if field_value is None or field_value == '':
+        return True
```

Comment:
All of them are used in validation too. x-nullable is used [here](https://github.com/zalando/connexion/blob/53908d30e566ccb81d73d3e27b461f407a8a8d31/connexion/utils.py#L101) . min/max length constraints are checked if the type is string. Format is used for special purpose or custom validation as per our use-case. However, the jsonschema designers made sure that if someone incorrectly specifies format for an invalid data type (eg. int instead of string), then perhaps it shouldn't throw a validation error. [examples](https://github.com/Julian/jsonschema/blob/ee2de6d0e0b6fed6f580cf2d744fb2790fe06a54/jsonschema/_format.py#L151)

As for the checking if value if None or an empty string, this is done because UI sends a blank string value, and since buildID is nullable field, hence allowing blank values to pass through the validation checks and then when processing them [here](https://github.com/aksareen/balrog/blob/first_set_of_apis_migrate/auslib/web/admin/views/rules.py#L56) , we can simply pass them into DB as NULL.